core/node/rpc/sync/client/syncer_set.go

```diff
@@ -737,6 +763,9 @@ func (ss *SyncerSet) selectNodeForStream(ctx context.Context, cookie *SyncCookie
 		for range remotes {
 			if _, err = ss.getOrCreateSyncer(ctx, selectedNode); err == nil {
 				return selectedNode, true
+			} else {
+				logging.FromCtx(ss.globalCtx).Errorw("Failed to get or create syncer for remote node",
```

Comment:
It is an idea to start a sub span for each remote. That way we can get details information what remote is slow or has issues.